backend/src/main/java/ai/verta/modeldb/project/subtypes/CreateProjectHandler.java

```diff
@@ -0,0 +1,192 @@
+            .addAllArtifacts(
+                request.getArtifactsList().stream()
+                    .sorted(Comparator.comparing(Artifact::getKey))
+                    .collect(Collectors.toList()))
+            .setReadmeText(request.getReadmeText())
+            .setCustomPermission(request.getCustomPermission())
+            .setVersionNumber(1L);
+
+    if (request.getDateCreated() != 0L) {
+      projectBuilder
+          .setDateCreated(request.getDateCreated())
+          .setDateUpdated(request.getDateCreated());
+    } else {
+      projectBuilder
+          .setDateCreated(Calendar.getInstance().getTimeInMillis())
+          .setDateUpdated(Calendar.getInstance().getTimeInMillis());
+    }
+
+    return projectBuilder.build();
+  }
+
+  public InternalFuture<Project> insertProject(Project newProject) {
+    final var now = Calendar.getInstance().getTimeInMillis();
+    Map<String, Object> valueMap = new LinkedHashMap<>();
+    valueMap.put("id", newProject.getId());
+    valueMap.put("name", newProject.getName());
+    valueMap.put("short_name", newProject.getShortName());
+    valueMap.put("description", newProject.getDescription());
+    valueMap.put("date_created", newProject.getDateCreated());
+    valueMap.put("date_updated", newProject.getDateUpdated());
+    valueMap.put("owner", newProject.getOwner());
+    valueMap.put("version_number", newProject.getVersionNumber());
+    valueMap.put("readme_text", newProject.getReadmeText());
+
+    valueMap.put("deleted", false);
+    valueMap.put("created", false);
+    valueMap.put("visibility_migration", true);
+
+    return jdbi.useHandle(
+            handle ->
+                handle.useTransaction(
+                    TransactionIsolationLevel.SERIALIZABLE,
+                    handleForTransaction -> {
+                      String queryString = buildInsertQuery(valueMap, "project");
+
+                      LOGGER.trace("insert project query string: " + queryString);
+                      var query = handleForTransaction.createUpdate(queryString);
+
+                      // Inserting fields arguments based on the keys and value of map
+                      for (Map.Entry<String, Object> objectEntry : valueMap.entrySet()) {
+                        query.bind(objectEntry.getKey(), objectEntry.getValue());
+                      }
+
+                      try {
+                        int count = query.execute();
+                        LOGGER.trace("Project Inserted : " + (count > 0));
+                      } catch (UnableToExecuteStatementException exception) {
+                        // take a brief pause before resubmitting its query/transaction
+                        Thread.sleep(config.getJdbi_retry_time()); // Time in ms
+                        LOGGER.trace("Retry to insert Project");
+                        int count = query.execute();
+                        LOGGER.trace("Project Inserted after retry : " + (count > 0));
+                      }
```

Comment:
Don't we already have a retry mechanism we can re-use?

Reply:
No we can not use that mechanism here because here exception is deferent and use case is also deferent.

Reply:
it might be thrown when some DB concurrency accord while project insertion then it will throw `UnableToExecuteStatementException` so at that time if we can take some milli sec pause then it will goes succeed. so i have added pause there instead of throwing error to the user.

Reply:
So what do you thing here? should we return error message to the user to do a retry?

Reply:
Removed pause and implemented retry logic using InternalFuture